src/main/java/com/mojang/brigadier/tree/ArgumentCommandNode.java

```diff
@@ -56,7 +57,16 @@ public String getUsageText() {
     @Override
     public void parse(final StringReader reader, final CommandContextBuilder<S> contextBuilder) throws CommandSyntaxException {
         final int start = reader.getCursor();
-        final T result = type.parse(reader);
+        T result = type.parse(reader);
+
+        if (result == null) {
+            result = type.parse(reader, contextBuilder.getSource());
```

Comment:
I think the more specific context should be attempted first, using the non-context method as a fallback.

Reply:
I implemented it this way to have the original purpose (i.e. minecraft) take the least complex path.

Reply:
Comment should be marked as unresolved, no discussion has occurred and Minecraft would still operate correctly under this case.